cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -490,12 +489,17 @@ public ApplicationWithPrograms deployApp(NamespaceId namespace, @Nullable String
   /**
    * Remove all the applications inside the given {@link Id.Namespace}
    *
-   * @param namespaceId the {@link Id.Namespace} under which all application should be deleted
+   * @param namespaceId the {@link NamespaceId} under which all application should be deleted
    * @throws Exception
    */
   public void removeAll(final NamespaceId namespaceId) throws Exception {
-    List<ApplicationSpecification> allSpecs = new ArrayList<>(
-      store.getAllApplications(namespaceId));
+    List<ApplicationSpecification> allSpecs = new ArrayList<>(store.getAllApplications(namespaceId));
+    Map<ApplicationId, ApplicationSpecification> apps = new HashMap<>();
+    for (ApplicationSpecification appSpec : allSpecs) {
+      ApplicationId applicationId = namespaceId.app(appSpec.getName(), appSpec.getAppVersion());
+      authorizationEnforcer.enforce(applicationId, authenticationContext.getPrincipal(), Action.ADMIN);
```

Comment:
This is tested here: https://github.com/caskdata/cdap/blob/release/4.3/cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java#L537